### PR TITLE
feat: make pagination easy to use

### PR DESCRIPTION
**Pull request by DorianMaliszewski**
## Type

- Feature

#### What is expected?

New pagination component. Need to refactor List to drop the old Pagination component

#### The following changes where made:

1. Simplify Pagination component

## Relevant logs and/or screenshots

Capture d’écran 2022-09-22 à 16 53 54

### CODE REVIEW

File: src/components/PaginationV2/index.tsx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+const PageNumbersContainer = styled.div`
+  display: flex;
+  gap: ${({ theme }) => theme.space['2']};
+  margin: 0 ${({ theme }) => theme.space['2']};
+`
+
+const StyledPageSwitch = styled(Button)`
+  width: ${({ theme }) => theme.space['4']};
+  height: ${({ theme }) => theme.space['6']};
```
Comment (matthprost) [inline src/components/PaginationV2/index.tsx:16]: please do not use space for sizing, it is reserved for spacing (padding, margin, etc)

File: src/components/PaginationV2/index.tsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+const PageSwitchContainer = styled.div`
+  display: flex;
+  gap: ${({ theme }) => theme.space['1']};
+`
+
+const StyledPageButton = styled.button`
+  color: ${({ theme }) => theme.colors.neutral.textStrong};
+  line-height: ${({ theme }) => theme.space['3']};
+  width: ${({ theme }) => theme.space['6']};
+  height: ${({ theme }) => theme.space['6']};
```
Comment (matthprost) [inline src/components/PaginationV2/index.tsx:31]: same here ?

File: src/components/PaginationV2/index.tsx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+`
+
+const PageSwitchContainer = styled.div`
+  display: flex;
+  gap: ${({ theme }) => theme.space['1']};
+`
+
+const StyledPageButton = styled.button`
+  color: ${({ theme }) => theme.colors.neutral.textStrong};
+  line-height: ${({ theme }) => theme.space['3']};
```
Comment (matthprost) [inline src/components/PaginationV2/index.tsx:29]: maybe you can use `theme.typography.body.lineHeight` instead

File: src/components/PaginationV2/index.tsx [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+  width: ${({ theme }) => theme.space['6']};
+  height: ${({ theme }) => theme.space['6']};
+  font-size: 16px;
+  font-weight: 500;
```
Comment (matthprost) [inline src/components/PaginationV2/index.tsx:33]: Same here can we use it from theme or use `Text` component ?

File: src/components/PaginationV2/index.tsx [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+    onChange(pageCount)
+  }
+
+  const goToNextPage = () => {
+    onChange(page + 1)
+  }
+
+  const goToPreviousPage = () => {
+    onChange(page - 1)
+  }
```
Comment (matthprost) [inline src/components/PaginationV2/index.tsx:104]: All of those functions call be `useCallBack` no ?

File: src/components/PaginationV2/index.tsx [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+
+  const pageNumbersToDisplay = useMemo(
+    () => (pageCount > 1 ? getPageNumbers(page, pageCount, pageTabCount) : [1]),
+    [page, pageCount, pageTabCount],
+  )
+
+  const handlePageClick = (pageNumber: number) => () => {
+    onChange(pageNumber)
+  }
```
Comment (matthprost) [inline src/components/PaginationV2/index.tsx:113]: this one too

File: src/components/PaginationV2/index.tsx [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+  )
+}
+
+Pagination.propTypes = {
+  disabled: PropTypes.bool,
+  onChange: PropTypes.func.isRequired,
+  page: PropTypes.number.isRequired,
+  pageCount: PropTypes.number.isRequired,
+  pageTabCount: PropTypes.number,
+}
```
Comment (matthprost) [inline src/components/PaginationV2/index.tsx:176]: can be removed we don't specify proptypes ad it is typescript

**Review by matthprost — CHANGES_REQUESTED** [review]

File: src/components/PaginationV2/getPageNumbers.ts [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    remaining = end - pageCount
+    end = pageCount
+  }
+  let start = currentPage - gap - remaining
+  if (start < 1) {
+    remaining = Math.abs(start - 1)
+    start = 1
+  }
+  if (end < pageCount) {
+    end = end + remaining > pageCount ? pageCount : end + remaining
```
Comment (QuiiBz) [inline src/components/PaginationV2/getPageNumbers.ts:26]:
```suggestion
    end = Math.min(end + remaining, pageCount)
```

File: src/components/PaginationV2/index.tsx [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+
+  const handlePageClick = useCallback(
+    (pageNumber: number) => () => {
+      onChange(pageNumber)
+    },
+    [onChange],
+  )
+
+  return (
+    <div style={{ display: 'flex' }}>
```
Comment (QuiiBz) [inline src/components/PaginationV2/index.tsx:118]: Maybe better to use `styled` instead?

**Review by baptisteArno — APPROVED** [review]
lgtm

**Pull request merged**